packages/jupyter-ai/src/components/copy-button.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useCallback } from 'react';
+
+import { Box, Button } from '@mui/material';
+
+enum CopyStatus {
+  None,
+  Copying,
+  Copied
+}
+
+const COPYBTN_TEXT_BY_STATUS: Record<CopyStatus, string> = {
+  [CopyStatus.None]: 'Copy To Clipboard',
+  [CopyStatus.Copying]: 'Copying...',
+  [CopyStatus.Copied]: 'Copied!'
```

Comment:
```suggestion
  [CopyStatus.None]: 'Copy to Clipboard',
  [CopyStatus.Copying]: 'Copying …',
  [CopyStatus.Copied]: 'Copied!'
```

Reply:
Actually, on second thought, do we need a separate string for the intermediate step of "Copying …"? The action is almost immediate, since there's no network connection or heavy computation, so on my PC, I see "Copy to Clipboard" switch to "Copying …" and then immediately to "Copied!". It is slightly distracting to have a one-frame intermediate state.

Reply:
Agree. Also presence of "Copying" also causes 1 additional rerender every CopyButton click without providing much value to the users. Removed the "Copying" status.